### DISCUSSION TITLE
Catching 'send_interactive' exception is subtask

### DISCUSSION THREAD

**Discussion by lucasmarcel** [scrapli/nornir_scrapli#103] [Q&A] [answered]
I'm using the send_interactive task in a Nornir subtask to reload switches.
It works, but as soon as the reload is confirmed, the connection is terminated by the target device and a ScrapliConnectionError is raised.
I wrapped task.run(send_interactive...) in a try/except block but I can only catch the NornirSubTaskError. I was hoping to handle the ScrapliConnectionError and log it has a 'success' since it's expected behavior.
Is there a way to do that?

**Comment by carlmontanari** [comment, marked as answer]
Honestly I don't use nornir at all so I don't know off hand :) 

It vaguely looks like the task should get stashed in the SubTaskError from looking at this https://github.com/nornir-automation/nornir/blob/develop/nornir/core/exceptions.py#L86-L88 and https://github.com/nornir-automation/nornir/blob/develop/nornir/core/task.py. Is that maybe helpful?

**Reply by lucasmarcel** [reply]
Thanks, yes I can work with that. I'll catch the Nornir exception, verify it contains the ScrapliConnectionError I'm expecting and log that.

**Reply by rvmcinnis** [reply]
hey @lucasmarcel  do you mind sharing your code you use to fix this? I guess the exception is still getting logged into the nornir.log ?

**Reply by momolemo** [reply]
Not the better solution but i have the same problem and to avoid this problem i dont reboot my switch instantly but i program the reboot after 1 minutes or more. so i can disconnect from the switch correctly. My HPE switch can schedule reboot maybe all switch/equipement cannot do this.
